src/main.js

```diff
@@ -0,0 +1,89 @@
+const EventEmitter = require('events')
+
+class SingleEmit extends EventEmitter {
+  constructor (eventEmitters, event) {
+    super()
+    if (typeof eventEmitters !== 'object' || eventEmitters === null)
+      throw new Error(
+        'Invalid type! The constructor expects an Array or an Object')
+    if (!eventEmitters.length && !Object.keys(eventEmitters).length)
+      throw new Error(
+        'The constructor expects a collection of items')
+    if (!event || !(typeof event === 'string' || typeof event === 'symbol'))
+      throw new Error(
+        'No event to listen to. See documentation')
+
+    this.emitted = false
+    if (Array.isArray(eventEmitters))
+      this._handleArray(eventEmitters, event)
+    else
+      this._handleObject(eventEmitters, event)
+  }
+
+  _handleArray (eventEmitters, event) {
+    const data = []
+    let emitted = 0
+
+    for (let i = 0; i < eventEmitters.length; i++) {
+      if (!(eventEmitters[i] instanceof EventEmitter)) {
+        SingleEmit._emitTypeError(this, eventEmitters[i])
+        break
+      }
+      eventEmitters[i].on('error', (error) => {
+        this._emitOnce('error', error)
+      })
+      eventEmitters[i].once(event, (eventData) => {
```

Comment:
why are you using "once" here?

Reply:
Only the first occurrence of the event should be recorded. Documented [here](https://github.com/wizzy25/single-emit/pull/4/files#diff-04c6e90faac2675aa89e2176d2eec7d8R52)